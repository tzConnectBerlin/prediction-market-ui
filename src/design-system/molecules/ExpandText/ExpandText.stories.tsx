import * as React from 'react';
// also exported from '@storybook/react' if you can deal with breaking changes in 6.1
import { Story, Meta } from '@storybook/react/types-6-0';
import { ExpandText, ExpandTextProps } from './ExpandText';

export default {
  title: 'Molecule/ExpandText',
  component: ExpandText,
  argTypes: {},
} as Meta;

const Template: Story<ExpandTextProps> = (args) => <ExpandText {...args} />;

export const LargeText = Template.bind({});
LargeText.args = {
  text: 'Lorem ipsum dolor sit amet consectetur adipisicing elit.'.repeat(20),
  expandActionText: 'Read more',
  shrinkActionText: 'Read less',
  hasExpandButton: true,
};

export const SmallText = Template.bind({});
SmallText.args = {
  text: 'Lorem ipsum dolor sit amet consectetur adipisicing elit.',
  expandActionText: 'Read more',
  shrinkActionText: 'Read less',
};
