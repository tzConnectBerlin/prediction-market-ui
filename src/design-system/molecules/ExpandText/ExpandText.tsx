import { useState } from 'react';
import styled from '@emotion/styled';
import { Typography } from '../../atoms/Typography';
import { CustomButton } from '../../atoms/Button';

export interface ExpandTextProps {
  text: string;
  expandActionText: string;
  shrinkActionText: string;
  handleToggleClick?: (value: boolean) => void | Promise<void>;
}

const CustomButtonStyled = styled(CustomButton)`
  padding-left: 0;
  padding-right: 0;
`;

export const ExpandText: React.FC<ExpandTextProps> = ({
  text,
  handleToggleClick,
  shrinkActionText,
  expandActionText,
}) => {
  const [isTruncated, setIsTruncated] = useState(false);
  const [truncate, setTruncate] = useState(true);
  const handleIsTruncate = (value: boolean) => {
    if (!isTruncated) {
      setIsTruncated(value);
    }
  };
  const handleToggle = () => {
    const newTruncate = !truncate;
    setTruncate(newTruncate);
    handleToggleClick && handleToggleClick(newTruncate);
  };
  return (
    <>
      <Typography truncate={truncate} isTruncated={handleIsTruncate}>
        {text}
      </Typography>
      {isTruncated && (
        <CustomButtonStyled
          variant="text"
          label={truncate ? expandActionText : shrinkActionText}
          onClick={handleToggle}
          disableElevation
          disableFocusRipple
          disableRipple
          disableTouchRipple
        />
      )}
    </>
  );
};
