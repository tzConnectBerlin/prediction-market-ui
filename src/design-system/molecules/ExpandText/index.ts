export { ExpandText, ExpandText as default } from './ExpandText';
export type { ExpandTextProps } from './ExpandText';
