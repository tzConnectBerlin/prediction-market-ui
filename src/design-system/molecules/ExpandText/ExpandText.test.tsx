import renderer from 'react-test-renderer';
import { ExpandText } from './ExpandText';

const largeTextArgs = {
  text: 'Lorem ipsum dolor sit amet consectetur adipisicing elit.'.repeat(20),
  expandActionText: 'Read more',
  shrinkActionText: 'Read less',
};

const smallTextArgs = {
  text: 'Lorem ipsum dolor sit amet consectetur adipisicing elit.',
  expandActionText: 'Read more',
  shrinkActionText: 'Read less',
};

// TODO: add testing for toggle click

describe('Snapshot testing ExpandText Component', () => {
  it('renders correctly with Large text', () => {
    const component = renderer.create(<ExpandText {...largeTextArgs} />).toJSON();
    expect(component).toMatchSnapshot();
  });

  it('renders correctly with small text', () => {
    const component = renderer.create(<ExpandText {...smallTextArgs} />).toJSON();
    expect(component).toMatchSnapshot();
  });
});
