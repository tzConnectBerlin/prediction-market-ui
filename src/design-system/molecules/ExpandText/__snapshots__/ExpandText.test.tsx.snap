// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot testing ExpandText Component renders correctly with Large text 1`] = `
.emotion-0 {
  margin: 0;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.5;
  letter-spacing: 0.00938em;
}

.emotion-0.truncate {
  overflow: hidden;
  display: -webkit-box;
  -webkit-box-orient: vertical;
  -webkit-line-clamp: 3;
}

<p
  className="MuiTypography-root MuiTypography-body1  truncate emotion-0"
  lines={3}
>
  Lorem ipsum dolor sit amet consectetur adipisicing elit.Lorem ipsum dolor sit amet consectetur adipisicing elit.Lorem ipsum dolor sit amet consectetur adipisicing elit.Lorem ipsum dolor sit amet consectetur adipisicing elit.Lorem ipsum dolor sit amet consectetur adipisicing elit.Lorem ipsum dolor sit amet consectetur adipisicing elit.Lorem ipsum dolor sit amet consectetur adipisicing elit.Lorem ipsum dolor sit amet consectetur adipisicing elit.Lorem ipsum dolor sit amet consectetur adipisicing elit.Lorem ipsum dolor sit amet consectetur adipisicing elit.Lorem ipsum dolor sit amet consectetur adipisicing elit.Lorem ipsum dolor sit amet consectetur adipisicing elit.Lorem ipsum dolor sit amet consectetur adipisicing elit.Lorem ipsum dolor sit amet consectetur adipisicing elit.Lorem ipsum dolor sit amet consectetur adipisicing elit.Lorem ipsum dolor sit amet consectetur adipisicing elit.Lorem ipsum dolor sit amet consectetur adipisicing elit.Lorem ipsum dolor sit amet consectetur adipisicing elit.Lorem ipsum dolor sit amet consectetur adipisicing elit.Lorem ipsum dolor sit amet consectetur adipisicing elit.
</p>
`;

exports[`Snapshot testing ExpandText Component renders correctly with small text 1`] = `
.emotion-0 {
  margin: 0;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.5;
  letter-spacing: 0.00938em;
}

.emotion-0.truncate {
  overflow: hidden;
  display: -webkit-box;
  -webkit-box-orient: vertical;
  -webkit-line-clamp: 3;
}

<p
  className="MuiTypography-root MuiTypography-body1  truncate emotion-0"
  lines={3}
>
  Lorem ipsum dolor sit amet consectetur adipisicing elit.
</p>
`;
